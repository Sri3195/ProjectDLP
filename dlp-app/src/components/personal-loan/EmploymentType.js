import { Card, Box, Typography, LinearProgress, Grid, RadioGroup, FormControlLabel, Radio, FormControl, FormLabel } from "@mui/material";
import { Link, useNavigate } from "react-router-dom";
import { useEffect } from "react";
import { useState } from "react";
import Button from "@mui/material/Button";
import "../../loans.css";
import ploan from "../../utils/personal-loan.jpg"
import ProgressBar from "../../ProgressBar";


function EmploymentType() {
    const navigate=useNavigate()

    const [employeetype,setEmployeeType]=useState('')
    
    const handleChange=(e)=>{
        setEmployeeType(e.target.value)
        
    }
    const url=new URL('http://localhost:8002/v1/update')
    const params={
        id: localStorage.getItem('phNo')
      }
      url.search = new URLSearchParams(params).toString();
   
    const data={
        id: localStorage.getItem('phNo'),
        phNo:localStorage.getItem('phNo'),
        empType: employeetype
    }
    const handleClick=()=>{
        if(employeetype===''){
            window.alert("Please select employment type !")
        }
        else{
            localStorage.setItem('empType',employeetype)
            fetch(url, {
                method: "PUT",
                mode: "cors",
                body: JSON.stringify(data),
                headers: {
                    "content-Type": "application/json",
                    "Access-Control-Allow-Origin": "*",
                }
            }).then(response => response.json())
            .then(json=>console.log(json))
            navigate("/personal-loan-income")
        }
        
    }
    return (
        <>
             <div style={{display:'flex'}}>
        <img src={ploan} alt="no-image found" style={{height: '100vh',width: '45%'}}/>
            <Box sx={{ mt: 5}}>
            <ProgressBar progress={0}/>
            <Card sx={{  ml: '80px',width: '600px', height: '550px', borderRadius: '10px' }}>
                    {/*<h3 style={{marginLeft:'10px',marginBottom:'50px'}}>Start your loan application by selecting employment type</h3>*/}

                    <FormControl>
                        <FormLabel  sx={{fontSize:'23px',mb:2,mt:5, color:'#5DADE2',ml:2,mr:10}}><center>Start your loan application by selecting Employment type</center></FormLabel>
                         <Grid sx={{mt:5}}>
                            <RadioGroup
                                row
                            >

                                <Grid item>
                                    <Card className="radios-emp"><FormControlLabel value="Salaried" control={<Radio />} label="Salaried" onChange={handleChange} /></Card>
                                </Grid>
                                <Grid item>
                                    <Card className="radios-emp"> <FormControlLabel value="Self-Employeed Business" control={<Radio />} label="Self-Employeed Business" onChange={handleChange}/></Card>
                                </Grid>
                                <Grid item>
                                    <Card className="radios-emp"><FormControlLabel value="Self-Employeed Professional" control={<Radio />} label="Self-Employeed Professional" onChange={handleChange}/></Card>
                                </Grid>
                            </RadioGroup>
                            <Button color="primary" variant="contained" sx={{ ml: 9, mt: 5, width: '410px', height: '50px' }} onClick={handleClick}>proceed</Button>
                        </Grid>
                    </FormControl>
                    
                </Card>
                
                    
                
            </Box>
            </div>
        </>
    )
}
export default EmploymentType;