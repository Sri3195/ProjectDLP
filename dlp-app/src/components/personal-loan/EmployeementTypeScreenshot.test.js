import React from "react";
import {createRenderer} from 'react-test-renderer/shallow'

import { MemoryRouter } from "react-router-dom";
import EmploymentType from './EmploymentType'
//create a test renderer
const renderer=createRenderer()

describe("Personal Loan Employment Type Screenshot test",()=>{
    test('renders correctly the personal loan employment type component',()=>{
        renderer.render(
            <MemoryRouter initialEntries={['/']}>
                <EmploymentType/>
          </MemoryRouter>
        );
        const result=renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });
});