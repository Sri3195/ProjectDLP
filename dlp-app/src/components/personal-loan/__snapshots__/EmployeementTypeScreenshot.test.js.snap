// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personal Loan Employment Type Screenshot test renders correctly the personal loan employment type component 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "createHref": [Function],
      "createURL": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "index": 0,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <EmploymentType />
</Router>
`;
